### DISCUSSION TITLE
Implementing Cache for GraphQL Response

### DISCUSSION THREAD

**Discussion by Abhi347** [octokit/octokit.js#2216] [Q&A]
Hi, I have Octokit setup for my Node.js server and I am calling graphQL queries us. The query response doesn't change mostly, but they are called every few seconds. So I was looking for a way to cache the response. I was thinking of throwing `node-cache` there for in-memory caching, instead of using something sophisticated like Redis, but before doing anything, I was thinking whether there's an inbuilt way to cache the response, especially for the servers.   
Any idea if we have something inbuilt or if someone has done it before, would you be willing to share your implementation?

**Comment by brignano** [comment]
I was also wondering the same thing.

I am currently using NextJS and can't seem to get a `cache: HIT`.

```
-  ┌ GET /api/user/repos 200 in 3949ms
   │
   ├──── GET https://github.com../api/v3/user/repos?per_page=100&af.. 200 in 1094ms (cache: MISS)
   │
   ├── 1 level ── GET https://github.com../api/v3/user/repos?per_page=100&af.. 200 in 803ms (cache: MISS)
   │
   ├── 1 level ── GET https://github.com../api/v3/user/repos?per_page=100&af.. 200 in 843ms (cache: MISS)
   │
   └── 1 level ── GET https://github.com../api/v3/user/repos?per_page=100&af.. 200 in 863ms (cache: MISS)
```

To accomplish this do we need to use an external library or is there something built into `octokit` that we can take advantage of so `cache: MISS` becomes `cache: HIT`?
